File: app/src/main/java/me/ianhenry/wordladder/models/Sound.java
package me.ianhenry.wordladder.models;

import android.media.MediaPlayer;

/**
 * Created by ianhe on 5/6/2017.
 */

public class Sound {
    public enum Type {
        FILE, TTS;
    }
    public static final String WELCOME = "Welcome";

    private Type type;
    private String text;
    private MediaPlayer.OnCompletionListener listener;

    public Sound(Type type, String text) {
        this(type, text, null);
    }

    public Sound(Type type, String text, MediaPlayer.OnCompletionListener listener) {
        this.type = type;
        this.text = text;
        this.listener = listener;
    }

    public Type getType() {
        return type;
    }

    public String getText() {
        return text;
    }

    public MediaPlayer.OnCompletionListener getListener() {
        return listener;
    }
}

File: app/src/main/java/me/ianhenry/wordladder/models/Sound.kt
package me.ianhenry.wordladder.models

import android.media.MediaPlayer

/**
 * Created by ianhe on 5/6/2017.
 */

class Sound @JvmOverloads constructor(val type: Type, val text: String, val listener: MediaPlayer.OnCompletionListener? = null) {
    enum class Type {
        FILE, TTS, TTS_PROMPT
    }

    companion object {
        val WELCOME = "Welcome"
    }
}
